### PR TITLE
if transaction status non-success, bypass intercept plugin

### PR DESCRIPTION
**Pull request by randall**
Don't allow the generator or statichit plugin to intercept the transaction
if a prior plugin has changed the transaction status to something non-success.

### CODE REVIEW

File: plugins/experimental/statichit/statichit.cc [hunk 1]
```diff
@@ -563,6 +563,12 @@ TSRemapInit(TSRemapInterface * /* api_info */, char * /* errbuf */, int /* errbu
 TSRemapStatus
 TSRemapDoRemap(void *ih, TSHttpTxn rh, TSRemapRequestInfo *rri)
 {
+  const TSHttpStatus txnstat = TSHttpTxnStatusGet(rh);
+  if (0 != static_cast<int>(txnstat) && txnstat != TS_HTTP_STATUS_OK) {
```
Comment (ywkaras) [inline plugins/experimental/statichit/statichit.cc:567]: Why not:
```
if (TS_HTTP_STATUS_NONE != txnstat && txnstat != TS_HTTP_STATUS_OK) {
```
?

Comment (randall) [reply plugins/experimental/statichit/statichit.cc:567]: Good point. That's what I get for copy/pasting.

Comment (randall) [reply plugins/experimental/statichit/statichit.cc]: Fixed. And added a bonus cleanup based on this feedback.

**Review by shinrich — APPROVED** [review]
Seems reasonable.

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Pull request merged**